apps/ewallet/lib/ewallet/web/orchestrator.ex

```diff
@@ -0,0 +1,67 @@
+defmodule EWallet.Web.Orchestrator do
+  @moduledoc """
+  This module should be used in every controller to deal with searching, filtering,
+  preloading and sorting.
+  """
+  alias EWallet.Web.{MatchAllParser, Paginator, Preloader, SearchParser, SortParser}
+
+  def query(query, overlay, attrs \\ nil) do
+    query
+    |> build_query(overlay, attrs)
+    |> Paginator.paginate_attrs(attrs)
+  end
+
+  def build_query(query, overlay, attrs \\ nil) do
+    query
+    |> preload_to_query(overlay, attrs)
+    |> MatchAllParser.to_query(attrs, overlay.filter_fields())
+    |> SearchParser.to_query(attrs, overlay.search_fields, default_mapped_fields())
+    |> SortParser.to_query(attrs, overlay.sort_fields, default_mapped_fields())
+  end
+
+  def all(records, overlay, attrs \\ nil)
+
+  def all(records, _overlay, %{"preload" => preload}) when is_map(preload) do
```

Comment:
Can we rename to `preload_all`? Reading it the first time, I was thinking if it's similar to [`Repo.all/2`](https://hexdocs.pm/ecto/Ecto.Repo.html#c:all/2), but it's about preloading.

Reply:
It "orchestrates" `all` or `one` - currently, it's true it only preloads, but we might add more things in the future and I don't want the controllers to necessarily know what's going on in the orchestrator. That's why I call it that way :)

Reply:
Thanks. Wasn't aware that it will expand so the current naming is fine to me.